manualsource/src/main/java/org/hibernate/search/manualsource/SearchFactory.java

```diff
@@ -0,0 +1,107 @@
+/*
+ * Hibernate Search, full-text search for your domain model
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+
+package org.hibernate.search.manualsource;
+
+import java.util.Set;
+
+import org.apache.lucene.analysis.Analyzer;
+
+import org.hibernate.search.indexes.IndexReaderAccessor;
+import org.hibernate.search.metadata.IndexedTypeDescriptor;
+import org.hibernate.search.query.dsl.QueryContextBuilder;
+import org.hibernate.search.stat.Statistics;
+
+/**
+ * Provide application wide operations as well as access to the underlying Lucene resources.
+ *
+ * @author Emmanuel Bernard
+ * @author Hardy Ferentschik
+ * @author Sanne Grinovero
+ */
+public interface SearchFactory {
+
+	/**
+	 * Optimize all indexes
+	 */
+	void optimize();
+
+	/**
+	 * Optimize the index holding {@code entityType}
+	 *
+	 * @param entityType the entity type (index) to optimize
+	 */
+	void optimize(Class<?> entityType);
+
+	/**
+	 * Retrieve an analyzer instance by its definition name
+	 *
+	 * @param name the name of the analyzer
+	 *
+	 * @return analyzer with the specified name
+	 *
+	 * @throws org.hibernate.search.exception.SearchException if the definition name is unknown
+	 */
+	Analyzer getAnalyzer(String name);
```

Comment:
Could we skip that? I understand it's copied essentially, but it'd be great to avoid the leakage of Lucene.